### PR TITLE
Create the JSON RPC endpoint to generate state sync proofs

### PR DESCRIPTION
**Pull request by vcastellm**
# Description

Create the JSON RPC endpoint to generate state sync proofs

This goes after:
* #964 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Querying the RPC endpoint

### CODE REVIEW

File: consensus/polybft/consensus_runtime.go [hunk 1]
```diff
@@ -856,6 +856,29 @@ func (c *consensusRuntime) GenerateExitProof(exitID, epoch, checkpointBlock uint
...
+	bundlesToExecute, err := c.state.getBundles(stateSyncID, 1)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get bundles: %w", err)
+	}
+
+	if len(bundlesToExecute) == 0 {
+		return nil, fmt.Errorf("cannot find bundle containing StateSync id %d", stateSyncID)
+	}
+
+	for _, bundle := range bundlesToExecute[0].StateSyncs {
```
Comment (goran-ethernal) [inline consensus/polybft/consensus_runtime.go:855]: Could we remove that `BundleProof` struct we had earlier, and when we build bundles, just directly build this new `StateSyncProof`, or will we do it in another PR?
It would even be easier to save these `StateSyncProof`s in `boltdb` and get them when `GetStateSyncProof` is called, because its key would be the `StateSync.ID`.

Comment (vcastellm) [reply consensus/polybft/consensus_runtime.go:855]: We are going to do that, in RPC we're only exposing StateSyncs, I can start right away with the bigger refactor after merging the complete workflow we have for the demo.

**Review by goran-ethernal — APPROVED** [review]
LGTM :100:

File: consensus/polybft/consensus_runtime.go [hunk 2]
```diff
@@ -856,6 +856,29 @@ func (c *consensusRuntime) GenerateExitProof(exitID, epoch, checkpointBlock uint
...
+		if bundle.ID == stateSyncID {
```
Comment (goran-ethernal) [inline consensus/polybft/consensus_runtime.go:856]: Small thing, I think we don't need this condition, since we have only one state sync event per bundle now, so it will contain only the one we are looking for.

Comment (vcastellm) [reply consensus/polybft/consensus_runtime.go:856]: Right, this was leftover from previous tests, but let's leave it as is at this time just in case, we'll remove it together with bundles in next PRs.

**Review by lazartravica — APPROVED** [review]

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by Stefan-Ethernal — APPROVED** [review]
